### PR TITLE
[Debt] Factor out bubbles

### PR DESCRIPTION
**Pull request by gmartin7**
Split all code using qtip between several files.

bloomQtipUtils has methods used by more than one function-type
bloomSourceBubbles contains MakeSourceTextBubbles and related code
bloomNotices contains PictureDictionary Experimental notice and "editing not allowed" messages
bloomHintBubbles contains bubbles created from <label> elements
plus DRT/LRT word markup was moved to readerTools
N.B.: other than bloomEditing.js all changes are to .ts files

### CODE REVIEW

File: src/BloomBrowserUI/bookEdit/js/bloomNotices.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+            if ($(this).css('cursor') == 'not-allowed') {
+                var whyDisabled = "You cannot change these because this is not the original copy.";
+                if ($(this).hasClass('bloom-readOnlyInEditMode')) {
+                    whyDisabled = "You cannot put anything in there while making an original book.";
+                }
+
+                var whatToSay = $(this).attr("data-hint");//don't use .data(), as that will trip over any } in the hint and try to interpret it as json
+
+                whatToSay = localizationManager.getLocalizedHint(whatToSay, $(this)) + " <br/>" + whyDisabled;
```
Comment (JohnThomson) [inline src/BloomBrowserUI/bookEdit/js/bloomNotices.ts]: Shouldn't "whyDisabled" be localized?

File: src/BloomBrowserUI/bookEdit/js/readerTools.js [hunk 2]
```diff
@@ -2,78 +2,122 @@
...
+        model.doMarkup();
+    });
+    $(container).find('.bloom-editable').focusin(function () {
+        model.noteFocus(this); // 'This' is the element that just got focus.
+    });
+    // and a slightly different one for keypresses
+    $(container).find('.bloom-editable').keypress(function () {
+        model.doKeypressMarkup();
+    });
+    $(container).find('.bloom-editable').keydown(function (e) {
```
Comment (JohnThomson) [inline src/BloomBrowserUI/bookEdit/js/readerTools.js]: Did we lose a comment somehow about why we need to do this rather obscure thing? (If not it's probably one I should have written, sorry.)

File: src/BloomBrowserUI/bookEdit/js/readerTools.ts [hunk 3]
```diff
@@ -42,10 +53,82 @@ function processDLRMessage(event: MessageEvent): void {
...
 
+function markDecodableStatus(): void {
+  // q-tips; mark sight words and non-decodable words
+  var editableElements = $(".bloom-content1");
+  editableElements.find('span.' + (<textMarkup>$).cssSightWord()).each(function() {
+    (<qtipInterface>$(this)).qtip({ content: 'Sight word' });
+  });
+
+  editableElements.find('span.' + (<textMarkup>$).cssWordNotFound()).each(function() {
+    (<qtipInterface>$(this)).qtip({ content: 'This word is not decodable in this stage.' });
```
Comment (JohnThomson) [inline src/BloomBrowserUI/bookEdit/js/readerTools.ts]: Function markDecodableStatus above seems to do the same thing. Do we need both?
Do we need localization here?

Comment (JohnThomson) [inline src/BloomBrowserUI/bookEdit/js/readerTools.ts]: Do we need localization here? (and similar places)

File: src/BloomBrowserUI/bookEdit/js/readerTools.ts [hunk 4]
```diff
@@ -42,10 +53,82 @@ function processDLRMessage(event: MessageEvent): void {
...
+  $(container).find('.bloom-editable').focusin(function () {
+    model.noteFocus(this); // 'This' is the element that just got focus.
+  });
+
+  // and a slightly different one for keypresses
+  $(container).find('.bloom-editable').keypress(function () {
+    model.doKeypressMarkup();
+  });
+
+  $(container).find('.bloom-editable').keydown(function (e) {
```
Comment (JohnThomson) [inline src/BloomBrowserUI/bookEdit/js/readerTools.ts]: Did we lose a comment somehow about why we need to do this rather obscure thing? (If not it's probably one I should have written, sorry.)

**Comment by JohnThomson** [comment]
I raised a few minor questions, probably not related to moving the code. It's really hard to tell whether any harm might be done by this sort of change because git is no help in figuring out what changed as opposed to just moving. Generally it looks good. @hatton, any reservations? Should we wait until post-beta or post-stable to merge this? I don't like leaving it hanging...

File: src/BloomBrowserUI/bookEdit/js/bloomNotices.ts (hunk 1 above)
Comment (gmartin7) [inline src/BloomBrowserUI/bookEdit/js/bloomNotices.ts]: Probably. I see "whatToSay" is localized, but not "whyDisabled". Existing code that I just moved, of course...

File: src/BloomBrowserUI/bookEdit/js/readerTools.ts (hunk 4 above)
Comment (gmartin7) [inline src/BloomBrowserUI/bookEdit/js/readerTools.ts]: I didn't see one. It seems to have to do with Undo/Redo, but I don't know much about why it needs to be *keydown*, if that's what you're asking.

File: src/BloomBrowserUI/bookEdit/js/readerTools.ts (hunk 3 above)
Comment (gmartin7) [inline src/BloomBrowserUI/bookEdit/js/readerTools.ts]: Fixing these decodable and leveled reader hints.

**Pull request merged**